cmd/skopeo/utils.go

```diff
@@ -257,3 +277,75 @@ func adjustUsage(c *cobra.Command) {
 	c.SetUsageTemplate(usageTemplate)
 	c.DisableFlagsInUseLine = true
 }
+
+func isRetryable(err error) bool {
+	err = errors.Cause(err)
+
+	if err == context.Canceled || err == context.DeadlineExceeded {
+		return false
+	}
+
+	type unwrapper interface {
+		Unwrap() error
+	}
+
+	switch e := err.(type) {
+
+	case unwrapper:
+		err = e.Unwrap()
+		return isRetryable(err)
+	case errcode.Error:
+		switch e.Code {
+		case errcode.ErrorCodeUnauthorized, errcodev2.ErrorCodeNameUnknown, errcodev2.ErrorCodeManifestUnknown:
+			return false
+		}
+		return true
+	case *net.OpError:
+		return isRetryable(e.Err)
+	case *url.Error:
+		return isRetryable(e.Err)
+	case syscall.Errno:
+		return e != syscall.ECONNREFUSED
+	case errcode.Errors:
+		// if this error is a group of errors, process them all in turn
+		for i := range e {
+			if !isRetryable(e[i]) {
+				return false
+			}
+		}
+		return true
+	case *multierror.Error:
+		// if this error is a group of errors, process them all in turn
+		for i := range e.Errors {
+			if !isRetryable(e.Errors[i]) {
+				return false
+			}
+		}
+		return true
+	}
+
+	return false
+}
+
+func retryIfNecessary(ctx context.Context, operation func() error, retryOptions *retryOptions) error {
+	err := operation()
+	for attempt := 0; err != nil && isRetryable(err) && attempt < retryOptions.maxRetry; attempt++ {
+		delay := func() time.Duration {
+			const base = 2
+			const initialRetryDelay = time.Second
+			return time.Duration(int(math.Pow(base, float64(attempt)))) * initialRetryDelay
+		}()
+
+		logrus.Infof("Warning: failed, retrying in %s ... (%d/%d)", delay, attempt+1, retryOptions.maxRetry)
```

Comment:
```suggestion
		logrus.Infof("Warning: failed, retrying in %s seconds ... (%d/%d)", delay, attempt+1, retryOptions.maxRetry)
```

Reply:
`delay` variable time.Duration type already has `s` as unit. I think it should be ok to use that?
```go
INFO[0000] Warning: failed, retrying in 1s 
```

Reply:
ooops, missed that.  Yes please do ignore this comment and TY!